kobo/apps/subsequences/actions/qual.py

```diff
@@ -254,10 +254,110 @@ def result_schema(self):
         }
         return schema
 
-    def get_output_fields(self):
-        return []
+    def get_output_fields(self) -> list[dict]:
+        output_fields = []
+        for qual_item in self.params:
+            field = {
+                'label': qual_item['labels']['_default'],
+                'source': self.source_question_xpath,
+                'name': f"{self.source_question_xpath}/{qual_item['uuid']}",
+                'type': qual_item['type'],
+                'dtpath': f"{self.source_question_xpath}/{qual_item['uuid']}",
+            }
+
+            if qual_item['type'] in ('qualSelectOne', 'qualSelectMultiple'):
+                field['choices'] = [
+                    {
+                        'uuid': choice['uuid'],
+                        'labels': choice['labels'],
+                    }
+                    for choice in qual_item.get('choices', [])
+                ]
+            output_fields.append(field)
+        return output_fields
 
     def transform_data_for_output(
-        self, action_data: list[dict]
-    ) -> dict[str, dict[str, Any]]:
-        return []
+        self, action_data: dict
+    ) -> dict[str, Any]:
+
+        qual_questions_by_uuid = {q['uuid']: q for q in self.params}
+
+        # Choice lookup tables for select questions
+        choices_by_uuid = {}
+        for qual_question in self.params:
+            if qual_question['type'] in ('qualSelectOne', 'qualSelectMultiple'):
+                choices_by_uuid[qual_question['uuid']] = {
+                    choice['uuid']: choice
+                    for choice in qual_question.get('choices', [])
+                }
+
+        results_list = []
+        for qual_uuid, qual_data in action_data.items():
+            if qual_uuid not in qual_questions_by_uuid:
+                continue
+
+            qual_question = qual_questions_by_uuid[qual_uuid]
+
+            # Get the most recent accepted version
+            versions = qual_data.get(self.VERSION_FIELD, [])
+            if not versions:
+                continue
+
+            # Get most recent accepted version
+            accepted_version = None
+            for version in versions:
+                if version.get(self.DATE_ACCEPTED_FIELD):
+                    accepted_version = version
+                    break
+
+            # Skip if no accepted version exists
+            if not accepted_version:
+                continue
+
+            # Extract the data and metadata
+            version_data = accepted_version.get(self.VERSION_DATA_FIELD, {})
```

Comment:
nit: this is confusing with all our other 'version' stuff
```suggestion
            selected_response_data = accepted_version.get(self.VERSION_DATA_FIELD, {})
```
Really we should probably rename 'VERSION_DATA_FIELD' but that's for another day